source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown

```diff
@@ -0,0 +1,290 @@
+
+
+## Introduction
+
+Daru is doing pretty good work as the data analysis & manipulation in IRuby notebook as well as backend part of web application. Ruby web application frameworks like Ruby on Rails, Sinatra, Nanoc are popular frameworks. So if Ruby developers get the gem like daru which can do data analysis
+and visualization work in applications, then there is no need of shifting to another language or usage of other gem.
+
+My project for <abbr title="Google Summer of Code 2017"> GSoC 2017</abbr> was to "make Daru more ready for integration with modern Web framework" in terms of visualization.
+
+To improve in terms of viewing data,
+[daru-view](https://github.com/Shekharrajak/daru-view), a plugin gem for
+[daru](https://github.com/SciRuby/daru) is created. [daru-view](https://github.com/Shekharrajak/daru-view) is for easy and interactive plotting in web application & IRuby notebook. It can work in frameworks like Rails, Sinatra, Nanoc and hopefully in others too.
+
+To see a quick overview of daru-view's features, have a look at these examples:
+
+* [IRuby Notebook Examples](http://nbviewer.jupyter.org/github/shekharrajak/daru-view/tree/master/spec/dummy_iruby/)
+
+* [daru io and daru-view usage in Rails app](https://github.com/Shekharrajak/daru_examples_io_view_rails)
+
+* [README of daru-view](https://github.com/Shekharrajak/daru-view/blob/master/README.md)
+
+
+## Examples
+
+This is how we can create a Plot class object:
+
+```ruby
+
+Daru::View::Plot.new(data, options)
+```
+
+* `data` can be `Daru::DataFrame`, data array.
+
+* `options` is a hash that contains various options to customize the chart.
+If you have chosen a plotting library then you must use the options according
+to the options the library providing. Here is the library `daru-view` uses.
+Please check the examples options, they are passing in javascript code:
+
+  * GoogleCharts: [https://developers.google.com/chart/interactive/docs/gallery](https://developers.google.com/chart/interactive/docs/gallery)
+
+  * HighCharts: [https://www.highcharts.com/demo](https://www.highcharts.com/demo)
+
+  * Nyaplot: [https://github.com/SciRuby/nyaplot](https://github.com/SciRuby/nyaplot) (it works same as `daru`)
+
+
+Note: User must have some knowledge about the plotting tool(that you want to
+use) to use it in `daru-view`. So that you can pass the correct options.
+
+
+### GoogleCharts:
+
+Set the plotting library to `:googlecharts` to use this adapter. This will
+load the required js files in your webpage or IRuby notebook.
+
+```ruby
+require 'daru/view'
+Daru::View.plotting_library = :googlecharts
+```
+
+Let's create a DataFrame :
+
+```
+idx = Daru::Index.new ['Year', 'Sales']
+data_rows = [
+          ['2004',  1000],
+          ['2005',  1170],
+          ['2006',  660],
+          ['2007',  1030]
+]
+df_sale_exp = Daru::DataFrame.rows(data_rows)
+df_sale_exp.vectors = idx
+```
+
+do some manupulation using `daru` if you want to do. Now time to plot it:
```

Comment:
Remove this line and add a comment at the bottom of the above code snippet:
``` ruby
# perform data manipulations....
```

Reply:
Just keep the `Now time to plot it:` part.

Reply:
Done.